### PR TITLE
Document usage of  `creduce`

### PR DESCRIPTION
**Pull request by michael-schwarz**
Part of #208.

### CODE REVIEW

File: docs/developer-guide/debugging.md [hunk 1]
```diff
@@ -117,3 +117,29 @@ code .           // Starts VS Code in the current directory
...
+    grep Fixpoint out.txt >/dev/null 2>&1
+else
+    exit 5
+fi
+```
+
+Note that Goblint exits with status `3` if the verifier fails.
+
+```console
+creduce --timeout 900 reduce.sh input.c
```
Comment (sim642) [inline docs/developer-guide/debugging.md:142]: Might be useful to also mention `--n` to speed things up considerably.

File: docs/developer-guide/debugging.md [hunk 2]
```diff
@@ -117,3 +117,29 @@ code .           // Starts VS Code in the current directory
...
+~/path/to/goblint input.c -v &> out.txt
+if [ $? -eq 3 ]; then
+    grep Fixpoint out.txt >/dev/null 2>&1
+else
+    exit 5
+fi
+```
+
+Note that Goblint exits with status `3` if the verifier fails.
+
```
Comment (sim642) [inline docs/developer-guide/debugging.md:140]: Should probably mention here that `./scripts/creduce/` contains additional more complex scripts.

Comment (michael-schwarz) [reply docs/developer-guide/debugging.md:140]: Damn, I didn't even know we had them :smile:

File: docs/developer-guide/debugging.md [hunk 3]
```diff
@@ -117,3 +117,31 @@ code .           // Starts VS Code in the current directory
 
 After VS Code has started, you can set breakpoints in the Goblint code. You can start debugging using the VS Code command palette with the command `"Debug: Start Debugging"`, or alternatively by pressing `F5`. Note that the Goblint execution is considerably slowed down in the debugger, so you have to be somewhat patient.
 Of course, don't forget to rebuild the debuggable executable if you make changes to the code!
+
+## Debugging Issues with Larger Programs
+
+Sometimes during development one may encounter instances where, e.g., the verifying phase reports that the fixpoint is not reached. This is usually due to bugs in `join`, `widen` or `leq`. For small programs, one can find the cause by inspecting the program and the output carefully. If the issue happens only with large programs it is hard to understand.
+
+To work on such cases, it makes sense to reduce the program to a small example program that still triggers the same issue. This can either be done by hand or using `creduce`. `creduce` takes as input a script that terminates with status `0` if the reduced program is still interesting and the program.
```
Comment (jerhard) [inline docs/developer-guide/debugging.md:125]: >if the reduced program is still interesting and the program.

There seems to be something missing?

Comment (michael-schwarz) [reply docs/developer-guide/debugging.md:125]: no, the sentence is just a bit awkward. It gets as arguments the script and the original program.

Comment (jerhard) [reply docs/developer-guide/debugging.md]: Ah, I see.

**Review by jerhard — APPROVED** [review]

**Pull request merged**